### ISSUE TITLE
Visual testing with Cypress

### ISSUE THREAD

**Issue by masakudamatsu** [masakudamatsu/line-height-picker#163]
Read https://docs.cypress.io/guides/tooling/visual-testing.html#Functional-vs-visual-testing for how to do it.

**Comment by masakudamatsu** [comment]
We tried Cypress Image Snapshot
- Official: https://www.npmjs.com/package/cypress-image-snapshot
- Docs: https://medium.com/norwich-node-user-group/visual-regression-testing-with-cypress-io-and-cypress-image-snapshot-99c520ccc595
But I keep getting two types of random errors.
- "The value of "offset" is out of range. It must be >= 0 and <= 13113564. Received -6792", which happens for the logo in 320px (but not in 1080px) and for buttons in 1080px (but not in 320px)
- "Image size (588x1575) different than saved snapshot size (589x172).", which happens to footer. in both 320px and 1080px, even though the previous snapshots were deleted.
We have no idea how to deal with these two errors.

**Comment by masakudamatsu** [comment]
It seems visual regression testing requires some paid software such as Applitools
- See https://blog.kevinlamping.com/visual-regression-testing-is-stupid/

**Comment by masakudamatsu** [comment]
See my tweet https://twitter.com/masa_kudamatsu/status/1273263197047279616?s=21 on how to deal with the errors

**Comment by masakudamatsu** [comment]
We've figured out how to take the full-page snapshots
- See https://github.com/palmerhq/cypress-image-snapshot/issues/78#issuecomment-491899161

We've created shortcut CLI commands to run cypress-image-snapshot
- `npm run test:visual` to run visual-tests.js only with Cypress CLI
- `npm run test:update` to update the snapshots

We only take the full-page snapshots
- This goes against the best practice of visual regression testing
    - https://blog.kevinlamping.com/visual-regression-testing-is-stupid/
    - https://docs.cypress.io/guides/tooling/visual-testing.html#DOM-state
- But we're interested in the overall layout, not each single component.
- We'll see if there's any inconvenience.

We take snapshots at each of the breakpoints as well as the narrowest screen (320px)

Snapshots are committed with Git LFS.
